### DISCUSSION TITLE
Allow manual World seed input

### DISCUSSION THREAD

**Discussion by SamAurel** [maruohon/minihud#188] [Ideas]
When playing on servers, players often dont have perms to /seed etc. Please allow a manual input in the configs for slime chunks and structure boxes.

**Comment by maruohon** [comment]
You can input the seed via a regular chat message (not a slash command) in the format `minihud-seed 12345`.
The message will not get sent to the server (assuming you don't make a typo in the first word).

Actually speaking of that, maybe I should also allow `/minihud-seed 12345`, that way it would never appear in the server chat even if you make a typo. It would probably appear in the server console and logs as "unknown command bla bla" though.

**Reply by SamAurel** [reply]
Thanks that helped a lot.
Would it be possible to make it a line in general tho?
It already worked for me but I was scared af.
Happy holidays,
Sam

**Reply by maruohon** [reply]
The main reason why it's not in the config menu with the other "normal" configs, is because seeds are world-specific. If it was stored in a regular config option, then you could forget about updating it in a given world and end up accidentally using the wrong seed in other worlds. The mod does store the seed for a given world/dimension once you enter it via chat, so you only need to do it once per world. 

Now that being said, I will probably add a separate small screen for inputting the seed. I'm not sure how you would open that, maybe I'll only do it in the rewritten code base where the malilib actions are available to open such a screen without adding a dangling button randomly somewhere in the config menu.

**Reply by maruohon** [reply]
Actually, for now, I might just add a hotkey that opens a seed input popup text field, as an alternative to the chat message.

**Reply by SamAurel** [reply]
Thank you very much.
